File: jserver-cypress/src/main/java/com/github/gr3gdev/jserver/test/route/StopRoute.java
package com.github.gr3gdev.jserver.test.route;

import com.github.gr3gdev.jserver.annotations.RouteMapping;
import com.github.gr3gdev.jserver.http.RequestMethod;
import com.github.gr3gdev.jserver.route.RouteListener;

public class StopRoute {

    @RouteMapping(path = "/stop", method = RequestMethod.GET)
    public RouteListener exec() {
        // TODO
        return new RouteListener();
    }


}

File: jserver-cypress/src/main/kotlin/com/github/gr3gdev/jserver/test/route/StopRoute.kt
package com.github.gr3gdev.jserver.test.route

import com.github.gr3gdev.jserver.Server
import com.github.gr3gdev.jserver.route.HttpStatus
import com.github.gr3gdev.jserver.route.Response
import com.github.gr3gdev.jserver.route.RouteListener

object StopRoute {

    fun exec(server: Server) = RouteListener().process {
        server.stop()
        Response(HttpStatus.OK)
    }

}